lib/Conversion/TTIRToTTIRDecomposition/TTIRToTTIRDecompositionPass.cpp

```diff
@@ -50,6 +50,10 @@ struct TTIRToTTIRDecompositionPass
     target.addIllegalOp<ttir::ConvolutionOp>();
     target.addIllegalOp<ttir::GetDimensionSizeOp>();
     target.addIllegalOp<ttir::PoolingOp>();
+    target.addDynamicallyLegalOp<ttir::ReshapeOp>([](ttir::ReshapeOp op) {
```

Comment:
Is there a need for this? Do you not want to design it in such a way that this reshape op is either deleted specifically or removed by remove-dead-values pass?

Reply:
As I have implemented the pattern inside the `TTIRtoTTIRDecompositionPass`, I have to mark the ReshapeOp conditionally illegal, so that the pattern will actually try to match the op. The other possible solution in to make a transform pass from scratch, as was done in https://github.com/tenstorrent/tt-mlir/pull/1252. Since we currently do not have a dedicated transformations pass, I put it here, but I am open to changing that and making a transformations pass folder.

Reply:
Update: In talks with @sdjordjevicTT, he has pointed out the existence of the folder for transform passes for TTIR ops, so I have changed my pass to live there.